app/src/main/java/com/github/onlynotesswent/model/file/FileRepositoryFirebaseStorage.kt

```diff
@@ -56,30 +58,44 @@ class FileRepositoryFirebaseStorage(private val db: FirebaseStorage) : FileRepos
     getFileRef(uid, fileType)
         .getFile(localFile)
         .addOnSuccessListener { onSuccess(localFile) }
-        .addOnFailureListener {
-          Log.e(TAG, "Error downloading image: ${it.message}", it)
-          onFailure(it)
+        .addOnFailureListener { error ->
+          val errorCode = (error as StorageException).errorCode
+          if (errorCode == StorageException.ERROR_OBJECT_NOT_FOUND) {
+            // Check should be external, since notes have a hasPdf and hasText field
+            Log.e(TAG, "File not found.", error)
+            onFileNotFound()
+          } else {
+            Log.e(TAG, "Error downloading file.", error)
+            onFailure(error)
+          }
         }
   }
 
   override fun deleteFile(
       uid: String,
       fileType: FileType,
       onSuccess: () -> Unit,
+      onFileNotFound: () -> Unit,
       onFailure: (Exception) -> Unit
   ) {
     getFileRef(uid, fileType)
         .delete()
         .addOnSuccessListener { onSuccess() }
-        .addOnFailureListener {
-          Log.e(TAG, "Error deleting image: ${it.message}", it)
-          onFailure(it)
+        .addOnFailureListener { error ->
+          val errorCode = (error as StorageException).errorCode
+          if (errorCode == StorageException.ERROR_OBJECT_NOT_FOUND) {
+            // Check should be external, since notes have a hasPdf and hasText field
+            Log.e(TAG, "File not found.", error)
+            onFileNotFound()
+          } else {
+            Log.e(TAG, "Error deleting file.", error)
+            onFailure(error)
+          }
         }
   }
 
   // Unnecessary as uploading with the same name will overwrite the image, but we can add it for
-  // clarity
-  // and possible future additions/implementations
+  // clarity and possible future additions/implementations
```

Comment:
Nice idea. Implementation is weird, as there is no simple check, instead I try to get a downloadURL, and check if it fails (following [this blog](https://jorgevergara.co/blog/storage-file-exist/#:~:text=When%20you%20need%20to%20check,the%20file%2C%20or%20an%20error.))

Reply:
Implementation will be like this:

```
getFileRef(uid, fileType).downloadUrl
          .addOnSuccessListener {
              uploadFile(uid, fileUri, fileType, onSuccess, onFailure)
          }
          .addOnFailureListener { error ->
              val errorCode = (error as StorageException).errorCode
              if (errorCode == StorageException.ERROR_OBJECT_NOT_FOUND) {
                  Log.e(TAG, "File not found.", error)
                  onFileNotFound()
              }
          }
```
Not great, creates a dependency on downloadUrl which could also fail separately, which doesn't mean that upload should fail, and it's an ugly work around. If you find a better way then great, otherwise I think it's best to code this check for other implementations if they have a difference between upload and update

Reply:
True. You can implement the same solution with metadata instead of downloadUrl, you still have the dependency on getMetadata (which is where metadata filed comes from) but checking metadata makes more sense then getting a download url.

Reply:
We will discuss with other's and if needed in the future modify updateFile to check if the file exists.

Reply:
True. I'll leave this for future consideration, since reading metadata correctly will lead to an additional billable metric being charged (a read).